components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/device/dao/DeviceFlowDAOImpl.java

```diff
@@ -0,0 +1,328 @@
+            try (PreparedStatement prepStmt =
+                         connection.prepareStatement(SQLQueries.DeviceFlowDAOSQLQueries.GET_SCOPE_FOR_USER_CODE)) {
+                ResultSet resultSet = null;
+                prepStmt.setString(1, userCode);
+                resultSet = prepStmt.executeQuery();
+                while (resultSet.next()) {
+                    scope = resultSet.getString(1);
+                }
+            }
+        } catch (SQLException e) {
+            throw new IdentityOAuth2Exception("Error when getting scopes for user code: " +
+                    userCode, e);
+        }
+        return scope;
+    }
+
+    @Override
+    public String getStatusForUserCode(String userCode) throws IdentityOAuth2Exception {
+
+        try (Connection connection = IdentityDatabaseUtil.getDBConnection(false)) {
+            try (PreparedStatement prepStmt =
+                connection.prepareStatement(SQLQueries.DeviceFlowDAOSQLQueries.GET_USER_CODE_STATUS)) {
+                ResultSet resultSet = null;
+                prepStmt.setString(1, userCode);
+                resultSet = prepStmt.executeQuery();
+                while (resultSet.next()) {
+                    status = resultSet.getString(1);
+                }
+            }
+        } catch (SQLException e) {
+            throw new IdentityOAuth2Exception("Error when getting status for user code: " + userCode, e);
+        }
+        return status;
+    }
+
+    @Override
+    public void setLastPollTime(String deviceCode, Timestamp newPollTime) throws IdentityOAuth2Exception {
+
+        try (Connection connection = IdentityDatabaseUtil.getDBConnection(true)) {
+            try (PreparedStatement prepStmt =
+                     connection.prepareStatement(SQLQueries.DeviceFlowDAOSQLQueries.SET_LAST_POLL_TIME)) {
+                prepStmt.setTimestamp(1, newPollTime, Calendar.getInstance(TimeZone
+                        .getTimeZone(Constants.UTC)));
+                prepStmt.setString(2, deviceCode);
+                prepStmt.execute();
+                IdentityDatabaseUtil.commitTransaction(connection);
+            } catch (SQLException e) {
+                IdentityDatabaseUtil.rollbackTransaction(connection);
+                throw new IdentityOAuth2Exception("Error when setting last poll time for device code: "
+                        + deviceCode, e);
+            }
+        } catch (SQLException e) {
+            throw new IdentityOAuth2Exception("Error when setting last poll time for device code: " + deviceCode, e);
+        }
+    }
+
+    @Override
+    public void setAuthzUser(String userCode, String userName) throws IdentityOAuth2Exception {
+
+        try (Connection connection = IdentityDatabaseUtil.getDBConnection(true)) {
+            try (PreparedStatement prepStmt =
+                         connection.prepareStatement(SQLQueries.DeviceFlowDAOSQLQueries.SET_AUTHZ_USER)) {
+                prepStmt.setString(1, userName);
```

Comment:
We need to store all user details not just the userName... We should store userName, domain, tenant, federatedIdp.. refer authz code.